src/Stacktrace.php

```diff
@@ -115,8 +115,10 @@ public function addFrame(string $file, int $line, array $backtraceFrame): void
             $line = (int) $matches[2];
         }
 
-        if (isset($backtraceFrame['class'])) {
+        if (isset($backtraceFrame['class']) && isset($backtraceFrame['function'])) {
             $functionName = sprintf('%s::%s', $backtraceFrame['class'], $backtraceFrame['function']);
+        } elseif (isset($backtraceFrame['class'])) {
```

Comment:
In which case there can be the `class` key but not the `function` key? If there is no real reason for this `elseif` statement I would simply leave the following cases:

- `isset($backtraceFrame['class']) && isset($backtraceFrame['function'])`
- `isset($backtraceFrame['function'])`
- everything else

Reply:
You can check issue #303, for which this PR was created. The issue described there doesn't affect only version 1.x, but also 2.x. you can also see me commenting a case when the key function does not exist.

Reply:
It I’m not missing something I see no reason the `class` key should be present but not the `function` key. This is what I meant

Reply:
Yes, I know, but yet there are some cases where it happens. One of the cases is when a class includes another file that has errors in it, as per my example in the issue.

Reply:
You can check [this comment](https://github.com/getsentry/sentry-php/issues/303#issuecomment-470053718) for more details of a case when it happened to me.

Reply:
Sentry treats a string without the `::` as a function name and in this case instead we would (incorrectly) report the name of the class as the name of the function. Please remove this `elseif` statement.

Reply:
If I remove the `elseif` then the bug will return.

Reply:
No, it will execute the next `elseif` that won't match too and then it will stop in the `else` statement. To clarify, the code should look like:

```php
if (isset($backtraceFrame['class']) && isset($backtraceFrame['function'])) {
    $functionName = sprintf('%s::%s', $backtraceFrame['class'], $backtraceFrame['function']);
} elseif (isset($backtraceFrame['function'])) {
    $functionName = $backtraceFrame['function'];
} else {
    $functionName = null;
}
```

In case the `class` key is present in `$backtraceFrame` but not the `function` key we should not report as function name the class name